### PR TITLE
Chat SDK setup

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

File: chatSDK/chat/src/androidTest/kotlin/com/walletconnect/chat/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+    @Test
+    fun useAppContext() {
+        // Context of the app under test.
+        val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+        assertEquals("com.walletconnect.chat.test", appContext.packageName)
+    }
+}
```
Comment (jakobuid) [inline chatSDK/chat/src/androidTest/kotlin/com/walletconnect/chat/ExampleInstrumentedTest.kt:24]: If not needed lets remove?

File: chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    fun reject(reject: Chat.Params.Reject, onError: (Chat.Model.Error) -> Unit)
+    fun message(message: Chat.Params.Message, onError: (Chat.Model.Error) -> Unit)
+    fun ping(ping: Chat.Params.Ping, onError: (Chat.Model.Error) -> Unit)
+    fun leave(leave: Chat.Params.Leave, onError: (Chat.Model.Error) -> Unit)
+    fun addContact(addContact: Chat.Params.AddContact, onError: (Chat.Model.Error) -> Unit)
+    fun getInvites(getInvites: Chat.Params.GetInvites): Map<String, Chat.Model.Invite>
+    fun getThreads(getThreads: Chat.Params.GetThreads): Map<String, Chat.Model.Thread>
+    fun getMessages(getMessages: Chat.Params.GetMessages): List<Chat.Model.Message>
+/***
+ TODO: Relay should be extracted to core along with websocket interface
```
Comment (jakobuid) [inline chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt:26]: 👍

File: chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package com.walletconnect.chat.client
+
+import com.walletconnect.chat.engine.domain.EngineInteractor
+import org.koin.core.KoinApplication
+
+internal class ChatProtocol : ChatInterface {
+    private val wcKoinApp: KoinApplication = KoinApplication.init()
+    private lateinit var engineInteractor: EngineInteractor
```
Comment (jakobuid) [inline chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt:8]: Maybe we could change `EngineInteractor` to `ChatEngine` and in sign module to `SignEngine`?
What do you guys think? @WalletConnect/kotlin-team

**Review by jakobuid — CHANGES_REQUESTED** [review]

**Review by jakobuid — APPROVED** [review]

File: chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt (hunk 3 above)
Comment (Elyniss) [reply chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt:8]: I agree.

Comment (TalhaAli00) [reply chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatProtocol.kt:8]: I like this idea

File: chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt (hunk 2 above)
Comment (TalhaAli00) [reply chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt:26]: For now, can we duplicate the code into Chat? I'm curious if Relay will change over time with the Chat SDK work

Comment (Elyniss) [reply chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt:26]: If we just copy code and create another instance we would have two websockets connections, right? For MVP I'm not against it but later on we should at investigate the solution

Comment (jakobuid) [reply chatSDK/chat/src/main/kotlin/com/walletconnect/chat/client/ChatInterface.kt:26]: For MVP it is good enough

**Pull request merged**